provider/marathon/marathon.go

```diff
@@ -421,33 +546,36 @@ func (p *Provider) hasHealthCheckLabels(application marathon.Application) bool {
 }
 
 func (p *Provider) getHealthCheckPath(application marathon.Application) string {
-	if label, ok := p.getLabel(application, types.LabelBackendHealthcheckPath); ok {
+	if label, ok := p.getAppLabel(application, types.LabelBackendHealthcheckPath); ok {
 		return label
 	}
 	return ""
 }
 
 func (p *Provider) getHealthCheckInterval(application marathon.Application) string {
-	if label, ok := p.getLabel(application, types.LabelBackendHealthcheckInterval); ok {
+	if label, ok := p.getAppLabel(application, types.LabelBackendHealthcheckInterval); ok {
 		return label
 	}
 	return ""
 }
 
-func (p *Provider) getBasicAuth(application marathon.Application) []string {
-	if basicAuth, ok := p.getLabel(application, types.LabelFrontendAuthBasic); ok {
+func (p *Provider) getBasicAuth(application marathon.Application, serviceName string) []string {
+	if basicAuth, ok := p.getLabel(application, types.LabelFrontendAuthBasic, serviceName); ok {
 		return strings.Split(basicAuth, ",")
 	}
 
 	return []string{}
 }
 
-func processPorts(application marathon.Application, task marathon.Task) (int, error) {
-	if portLabel, ok := (*application.Labels)[types.LabelPort]; ok {
-		port, err := strconv.Atoi(portLabel)
+//processPorts validates `portValue` to be valid numeric port.
+// if it is empty, proceeds to extract all ports and attempts to retrieve a port specified by the 'portIndexValue', given that it is a valid integer.
+// if all fails, returns first port (at index 0) from the list.
+func processPorts(application marathon.Application, task marathon.Task, portValue string, portIndexValue string) (int, error) {
```

Comment:
Could you please shorten the specification of the last two function parameters like `portValue, portIndexValue string`.